allure-framework-modules/allure-webapi-module/src/main/java/com/capgemini/ntc/example/core/base/driver/DriverManager.java

```diff
@@ -0,0 +1,86 @@
+package com.capgemini.ntc.example.core.base.driver;
+
+import com.capgemini.ntc.example.core.base.properties.PropertiesFileSettings;
+import com.capgemini.ntc.example.core.base.runtime.RuntimeParameters;
+import com.capgemini.ntc.test.core.logger.BFLogger;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+
+public class DriverManager {
+	
+	private static ThreadLocal<String> drivers = new ThreadLocal<String>();
+	
+	private static PropertiesFileSettings propertiesFileSettings;
+	
+	@Inject
+	public DriverManager(@Named("properties") PropertiesFileSettings propertiesFileSettings) {
+		
+		if (null == DriverManager.propertiesFileSettings) {
+			DriverManager.propertiesFileSettings = propertiesFileSettings;
+		}
+		
+		this.start();
+	}
+	
+	public void start() {
+		DriverManager.getDriver();
+	}
+	
+	public void stop() {
+		try {
+			closeDriver();
+			BFLogger.logDebug("Closing Driver in stop()");
+		} catch (Exception e) {
+		}
+	}
+	
+	@Override
+	protected void finalize() throws Throwable {
```

Comment:
What is the point of catching one exception and throwing another one within one method?